internal/host/validator.go

```diff
@@ -417,271 +388,214 @@ func (v *validator) areTangServersReachable(c *validationContext) ValidationStat
 	case ValidationFailure:
-		return fmt.Sprintf("The host is not eligible to participate in Openshift Cluster because the minimum required RAM for any role is %s, found only %s",
-			conversions.BytesToString(conversions.MibToBytes(c.minRAMMibRequirement)), conversions.BytesToString(c.inventory.Memory.PhysicalBytes))
-	case ValidationPending:
-		return "Missing inventory"
+		// Mode TPMv2
+		if c.inventory.TpmVersion == models.InventoryTpmVersionNone {
+			return status, "TPM version could not be found, make sure TPM is enabled in host's BIOS"
+		}
+		if c.cluster.DiskEncryption.Mode == nil {
+			return status, "Invalid LUKS object in ignition - both TPM2 and Tang are not available"
+		}
+		return status, fmt.Sprintf("The host's TPM version is not supported, expected-version: %s, actual-version: %s",
+			models.InventoryTpmVersionNr20, c.inventory.TpmVersion)
 	default:
-		return fmt.Sprintf("Unexpected status %s", status)
+		return status, fmt.Sprintf("Unexpected status %s", status)
 	}
 }
 
-func (v *validator) hasMinValidDisks(c *validationContext) ValidationStatus {
+func (v *validator) hasMinValidDisks(c *validationContext) (ValidationStatus, string) {
 	if c.inventory == nil {
-		return ValidationPending
+		return ValidationPending, "Missing inventory"
 	}
-
 	disks := v.hwValidator.ListEligibleDisks(c.inventory)
-	return boolValue(len(disks) > 0)
-}
-
-func (v *validator) printHasMinValidDisks(c *validationContext, status ValidationStatus) string {
-	switch status {
-	case ValidationSuccess:
-		return "Sufficient disk capacity"
-	case ValidationFailure:
-		return "No eligible disks were found, please check specific disks to see why they are not eligible"
-	case ValidationPending:
-		return "Missing inventory"
-	default:
-		return fmt.Sprintf("Unexpected status %s", status)
+	if boolValue(len(disks) > 0) == ValidationSuccess {
+		return ValidationSuccess, "Sufficient disk capacity"
 	}
+	return ValidationFailure, "No eligible disks were found, please check specific disks to see why they are not eligible"
 }
 
-func (v *validator) isMachineCidrDefined(c *validationContext) ValidationStatus {
-	if c.infraEnv != nil {
-		return ValidationSuccessSuppressOutput
-	}
-	return boolValue(swag.BoolValue(c.cluster.UserManagedNetworking) || swag.StringValue(c.cluster.Kind) == models.ClusterKindAddHostsCluster || network.IsMachineCidrAvailable(c.cluster))
-}
-
-func (v *validator) printIsMachineCidrDefined(context *validationContext, status ValidationStatus) string {
-	switch status {
-	case ValidationSuccess:
-		if swag.BoolValue(context.cluster.UserManagedNetworking) {
-			return "No Machine Network CIDR needed: User Managed Networking"
-		}
-		if swag.StringValue(context.cluster.Kind) == models.ClusterKindAddHostsCluster {
-			return "No Machine Network CIDR needed: Day2 cluster"
+func (v *validator) isMachineCidrDefined(c *validationContext) (ValidationStatus, string) {
+	status := ValidationSuccessSuppressOutput
+	if c.infraEnv == nil {
```

Comment:
I think this would be better reversed.
Then the rest of the function could be one level less indented.